compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -7087,4 +7091,32 @@ private void setClassQualifiers(NodeList<Token> qualifiers, BLangClassDefinition
             }
         }
     }
+
+    private List<BLangXMLStepExtend> createBLangXMLStepExtends(NodeList<Node> nodes, BLangExpression expr) {
+        List<BLangXMLStepExtend> extensions = new ArrayList<>();
```

Comment:
Since we add a value per each node let's intialize this ArrayList with correct size 
```suggestion
        List<BLangXMLStepExtend> extensions = new ArrayList<>(nodes.size());
```